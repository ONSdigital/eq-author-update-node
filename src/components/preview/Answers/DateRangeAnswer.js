import React from "react";
import PropTypes from "prop-types";

import DateAnswer from "./DateAnswer";

const DateRangeAnswer = ({ answer }) => {
  const { secondaryLabel, ...dateAnswer } = answer;

  return (
    <div>
      <DateAnswer answer={dateAnswer} />
      <DateAnswer
        answer={{
          ...dateAnswer,
          label: secondaryLabel,
        }}
      />
    </div>
  );
};
DateRangeAnswer.propTypes = {
  answer: PropTypes.shape({
    label: PropTypes.string,
    secondaryLabel: PropTypes.string,
    properties: PropTypes.shape({
      format: PropTypes.string,
    }).isRequired,
  }),
};

export default DateRangeAnswer;
