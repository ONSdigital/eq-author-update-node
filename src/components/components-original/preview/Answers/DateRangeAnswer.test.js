import { shallow } from "enzyme";
import React from "react";

import DateRangeAnswer from "./DateRangeAnswer";

describe("DateRangeAnswer", () => {
  const answer = {
    label: "from",
    secondaryLabel: "to",
    properties: {
      format: "dd/mm/yyyy",
    },
  };

  it("should render", () => {
    expect(shallow(<DateRangeAnswer answer={answer} />)).toMatchSnapshot();
  });
});
