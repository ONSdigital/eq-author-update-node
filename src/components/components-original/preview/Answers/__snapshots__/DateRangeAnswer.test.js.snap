// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeAnswer should render 1`] = `
<div>
  <DateAnswer
    answer={
      {
        "label": "from",
        "properties": {
          "format": "dd/mm/yyyy",
        },
      }
    }
  />
  <DateAnswer
    answer={
      {
        "label": "to",
        "properties": {
          "format": "dd/mm/yyyy",
        },
      }
    }
  />
</div>
`;
